bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleTTSService.java

```diff
@@ -332,6 +340,38 @@ public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFor
         if (audio == null) {
             throw new TTSException("Could not read from Google Cloud TTS Service");
         }
-        return new ByteArrayAudioStream(audio, requestedFormat);
+
+        // compute the real format returned by google
+        AudioFormat realAudioFormat = parseAudioFormat(audio);
+
+        return new ByteArrayAudioStream(audio, realAudioFormat);
+    }
+
+    private AudioFormat parseAudioFormat(byte[] audio) {
```

Comment:
Thanks for looking into this.

Two questions:
1. Should we wait for https://github.com/openhab/openhab-core/pull/2650 and use your new constructor for passing channels to the generated `AudioFormat`?
2. This is nearly the same code as used in https://github.com/openhab/openhab-core/pull/2650 and thus a util method in OHC makes a lot of sense to reduce duplicate code. Doesn't it?

Reply:
1 & 2 : Yes it seems logical. Good idea.
The googletts addon won't be compatible with the 3.2 branch tough. Is it an issue ?
I'm not familiar with "coupling" openhab-core and openhab-addons. Is there any special precaution to take ?

Reply:
With https://github.com/openhab/openhab-core/pull/2650 in place you might want to reconsider your code and use the new features and util classes in this PR.

Reply:
Thanks ! Done.
But the checks (continuous development build) failed.
Maybe I was too fast, and the change in openhab-core was not ready for openhab-addon ? If so, do you know when, and if I can trigger another build (apart from force pushing something) ?

Reply:
The code is built and published nightly. After a merge I OHC it will be available in the next day for openHAB Add-ons and other repositories.